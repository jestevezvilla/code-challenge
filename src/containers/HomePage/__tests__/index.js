import React from 'react';
import { shallow } from 'enzyme';

import { HomePage } from '../index';

describe('HomePage', () => {
  let props;
  let mountedComponent;
  const component = () => {
    if (!mountedComponent) {
      mountedComponent =
        shallow(<HomePage {...props} />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    props = {
      articles: [],
      fetchData() {},
      isFetching: true,
      onDeleteClick() {},
    };
    mountedComponent = undefined;
  });

  it('should snapshot be OK', () => {
    expect(component()).toMatchSnapshot();
  });

  it('should Loading when isFetchin', () => {
    expect(component().find('Loader').exists()).toBe(true);
  });
  
});
