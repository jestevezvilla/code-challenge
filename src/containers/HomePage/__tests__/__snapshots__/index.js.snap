// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should snapshot be OK 1`] = `
<div
  className="HomePage"
>
  <Loader
    title="Loading data..."
  />
</div>
`;
